src/Sentry.Unity.Editor/Native/BuildPostProcess.cs

```diff
@@ -40,43 +41,83 @@ public static void OnPostProcessBuild(BuildTarget target, string pathToProject)
-                //     return;
-                // }
-                //
-                // if (!sentryCliOptions.Validate(logger))
-                // {
-                //     logger.LogWarning("sentry-cli validation failed. Symbols will not be uploaded." +
-                //                        "\nYou can disable this warning by disabling the automated symbols upload under " +
-                //                        "Tools -> Sentry -> Editor");
-                //     return;
-                // }
-                //
-                // SentryCli.CreateSentryProperties(pathToProject, sentryCliOptions);
-                // SentryCli.AddExecutableToXcodeProject(pathToProject, logger);
-                //
-                // sentryXcodeProject.AddBuildPhaseSymbolUpload(logger);
             }
             catch (Exception e)
             {
-                logger.LogError("Failed to add the Sentry crash handler to the built application", e);
+                logger.LogError("Failed to add the Sentry native integration to the built application", e);
                 throw new BuildFailedException("Sentry Native BuildPostProcess failed");
             }
         }
+
+        private static void addCrashHandler(IDiagnosticLogger logger, string projectDir)
+        {
+            var crashpadPath = Path.GetFullPath(Path.Combine("Packages", SentryPackageInfo.GetName(), "Plugins",
+                "Windows", "Sentry", "crashpad_handler.exe"));
+            var targetPath = Path.Combine(projectDir, Path.GetFileName(crashpadPath));
+            logger.LogInfo($"Copying the native crash handler '{Path.GetFileName(crashpadPath)}' to the output directory");
+            File.Copy(crashpadPath, targetPath, true);
+        }
+
+        private static void uploadDebugSymbols(IDiagnosticLogger logger, string projectDir, string executableName)
+        {
+            var cliOptions = SentryCliOptions.LoadCliOptions();
+            if (!cliOptions.Validate(logger))
+                return;
+
+            var propertiesFile = SentryCli.CreateSentryProperties(projectDir, cliOptions);
+
+            logger.LogInfo($"Uploading debuging information using sentry-cli in {projectDir}");
+
+            var paths = "";
+            Func<string, bool> addPath = (string name) =>
+            {
+                var fullPath = Path.Combine(projectDir, name);
+                if (Directory.Exists(fullPath) || File.Exists(fullPath))
+                {
+                    paths += " " + name;
+                    logger.LogDebug($"Adding '{name}' to the debug-info upload");
+                    return true;
+                }
+                else
+                {
+                    logger.LogWarning($"Coudn't find '{name}' - debug symbol upload will be incomplete");
+                    return false;
+                }
+            };
+
+            addPath(executableName);
```

Comment:
Aren't we better off running sentry-cli on the root and let it find the relevant files? It'll also be more future proof if Unity introduces other native shared libraries

Reply:
I've thought about that but didn't like that it would also upload:
* UnityCrashHandler64.exe
* crashpad_handler.exe

Reply:
Do you mean it uploads the executables? My understanding was that `upload-dif` would only scan for and upload debug information files.

Reply:
executables also have some debug info. See for example here:

<img width="557" alt="image" src="https://user-images.githubusercontent.com/6349682/156821093-f78117a1-27a7-40f6-979b-58c79ee6549a.png">